packages/calypso-products/src/has-marketplace-product.ts

```diff
@@ -1,20 +1,17 @@
-const cleanSlug = ( slug: string ) =>
-	slug && slug.replace( /_/g, '-' ).split( /-(monthly|yearly|2y)/ )[ 0 ];
-
 /**
- * Returns true if a list of products contains a marketplace product with the specified product slug.
+ * Returns true if a list of products contains a marketplace product with a matching product or store product slug.
```

Comment:
Here you mention the store product slug but in code, it's named billing product slug. Could we avoid referring to the same concept with two different names?

Also, shouldn't the comment be:
`Returns true if a list of marketplace products contains either a subscription slug or billing product slug that matches the given product slug .`

Reply:
> Here you mention the store product slug but in code, it's named billing product slug. Could we avoid referring to the same concept with two different names?

It's not. The store product slug (legacy store) is subscriptionSlug, billing_product_slug is the slug from the billing catalog (not-legacy store), which by convention for marketplace plugins is the plugin slug.

> Returns true if a list of marketplace products contains either a subscription slug or billing product slug that matches the given product slug

It's a list of all products not just marketplace products but ok, my wording could be clearer

----

I definitely take your point about this being confusing, as the slugs usually are. 

I'll rename subscriptionSlug to storeProductSlug, as we don't call it 'subscriptionSlug' elsewhere that I know of, this should be clearer it's a familiar slug not a new one.

I'll rename productSlug to searchSlug, so that it's purpose is clearer.

I'll update this jsdoc to say something like:

Returns true if a list of products includes a product with a store product slug or billing product slug that matches the given slug.

I'll look at adding other comments too